terraformize/terraformize_terraform_wrapper.py

```diff
@@ -1,32 +1,38 @@
 from python_terraform import *
 from typing import Tuple, Optional
+import tempfile
+import shutil
 
 
 class Terraformize:
 
-    def __init__(self, workspace: str, folder_path: str, terraform_bin_path: Optional[str] = None):
+    def __init__(self, remote_backend: bool, workspace: str, folder_path: str, terraform_bin_path: Optional[str] = None):
         """
         Will create a terraform object, create a workspace & init the terraform directory
 
         Arguments:
+            :param remote_backend: if terraform is running with a remote state backend
             :param workspace: the workspace terraform will be executed in
             :param folder_path: the full path of the folder to run the terraform in
-            :param terraform_bin_path: the full path of the terraform binary to use, will try to use the one at the path
-            if not set
+            :param terraform_bin_path: the full path of the terraform binary to use, defaults to using $PATH if unset
         """
         self.terraform_bin_path = terraform_bin_path
-        self.tf = Terraform(working_dir=folder_path, terraform_bin_path=self.terraform_bin_path)
 
-        # we need to init for a remote backend before we can create a workspace
-        self.init_return_code, self.init_stdout, self.init_stderr = self.tf.init(dir_or_plan=folder_path)
+        if remote_backend is True:
+            self.working_path = tempfile.mkdtemp()
+            self.tf = Terraform(working_dir=self.working_path, terraform_bin_path=self.terraform_bin_path)
+            # using an envvar is the only way to prevent terraform from creating
+            # a default.tfstate during init on a remote backend (which causes locking)
+            os.environ["TF_WORKSPACE"] = workspace
```

Comment:
As per our talk at #18 we're sticking to file based workspace as the `TF_WORKSPACE` envvar doesn't guarantee creation of workspace on all backends (only select of pre-existing workspaces)